filebeat/registrar/registrar.go

```diff
@@ -176,6 +168,32 @@ func (r *Registrar) loadAndConvertOldState(f *os.File) bool {
 	return true
 }
 
+func convertOldStates(oldStates map[string]file.State) []file.State {
```

Comment:
Good point, that might be brittle and break if there are any changes to `StateOS`.